otherlibs/unix/mkdir.c

```diff
@@ -26,15 +26,16 @@
 #include <caml/signals.h>
 #include "caml/unixsupport.h"
 
-CAMLprim value caml_unix_mkdir(value path, value perm)
+CAMLprim value caml_unix_mkdir(value path, value vperm)
 {
-  CAMLparam2(path, perm);
+  CAMLparam2(path, vperm);
   char_os * p;
   int ret;
+  int perm = Int_val(perm);
```

Comment:
This "clearly correct" patch is clearly wrong, `Int_val(perm)` should be `Int_val(vperm)`.